### PR TITLE
Enable the generation of wheels for python 3.11 and upgrade python on CI/CD workflows to 3.8

### PR DESCRIPTION
**Pull request by AmintorDusko**
**Context:** This PR enables the generation of wheels for python 3.11. 
Also, we are updating the python version of CI/CD workflows from 3.7 to 3.8.

**Description of the Change:**

**Benefits:**
- Provide wheels for python 3.11.
- Faster CI/CD workflows with python 3.8.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Comment by AmintorDusko** [comment]
[sc-28724]

File: .github/workflows/build_wheel_manylinux2014.yml [hunk 1]
```diff
@@ -37,16 +37,17 @@ jobs:
 
     steps:
       - name: Cancel Previous Runs
-        uses: styfle/cancel-workflow-action@0.9.1
+        uses: styfle/cancel-workflow-action@0.10.0
```
Comment (multiphaseCFD) [inline .github/workflows/build_wheel_manylinux2014.yml:40]: The  `cancel-workflow-action` @0.11.0  was just released. Are we going to upgrade it to `0.11.0`?

**Comment by multiphaseCFD** [comment]
Hi @AmintorDusko, Nice work! Just leave on minor question.

File: .github/workflows/build_wheel_manylinux2014.yml (hunk 1 above)
Comment (AmintorDusko) [reply .github/workflows/build_wheel_manylinux2014.yml]: Thank you for reminding me of that.

**Review by mlxd — APPROVED** [review]
Nice work!

**Comment by multiphaseCFD** [comment]
Nice work @AmintorDusko !  Let me approve it.

**Review by multiphaseCFD — APPROVED** [review]

**Pull request merged**